src/network/HttpEndpoint.ts

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Endpoint } from "./AbstractEndpoint";
+import { URL } from "url";
+import { URIParseConfig } from "./Network.util";
+
+export class HttpEndpoint extends Endpoint {
+  constructor(url: string) {
+    super(url);
+
+    try {
+      this._parsedUrl = new URL(this.preprocessUri(url));
+      this._scheme = this._parsedUrl.protocol.replace(":", "");
+      this._hostname = this._parsedUrl.hostname.replace("[", "");
+      this._hostname = this._hostname.replace("]", "");
```

Comment:
Can there be multiple square brackets? [playground](https://www.typescriptlang.org/play?#code/G4QwTgBGCmELwQPQG0A6BdVz2IOYEsBuAKAGMB7AOwGdyAbaAOjvNwAoAiZAFwAt9qEARBARk1bmHyVc6dB0YwADnRClobGABoIHDgEp9ZKrQbNWnHv0HDR4ydNwLlq9Zug69h4zXpMW7Bx8wrYQElIycs7QKmoa2roG+kA)